File: Classwork/Classwork09/BinaryTree.java
/**
 *  File name: BinaryTree.java
 *  Purpose: Demonstrates a binary tree
 *  @author: B.J. Johnson
 *  @date: 2019-10-23
 *  @date: 2020-08-05: modified to use "Traversal" class with static methods
 */

import java.util.Scanner;

public class BinaryTree {

   public static void pressReturn() {
      System.out.print( "        [press 'enter' to see the result]... <=" );
      Scanner myInput = new Scanner( System.in );
      myInput.nextLine();
   }

   public static void main( String[] args ) {
      BinaryTreeNode root = new BinaryTreeNode( 0 );
      root.add( 1, "L" );
      root.add( 2, "R" );
      BinaryTreeNode one = root.getChild( "L" );
      BinaryTreeNode two = root.getChild( "R" );
      one.add( 3, "L" );
      one.add( 4, "R" );
      two.add( 5, "L" );
      two.add( 6, "R" );
      BinaryTreeNode six = two.getChild( "R" );
      six.add( 7, "L" );
      six.add( 8, "R" );

      System.out.println( "\n\n  ==== the tree looks like this [sorta]: \n\n" );
      System.out.println( "                   (0)" );
      System.out.println( "                   / \\" );
      System.out.println( "                  /   \\" );
      System.out.println( "                 /     \\" );
      System.out.println( "                /       \\" );
      System.out.println( "              (1)       (2)" );
      System.out.println( "              / \\       / \\" );
      System.out.println( "             /   \\     /   \\" );
      System.out.println( "           (3)   (4) (5)   (6)" );
      System.out.println( "                           / \\" );
      System.out.println( "                          /   \\" );
      System.out.println( "                        (7)   (8)" );

      System.out.println( "\n\n   Now doing a 'pre-order' traversal...\n" );
      pressReturn();
      System.out.print( "        " );
      Traversals.preOrderPrinter( root );
      System.out.println( "\n\n   Now doing a 'post-order' traversal...\n" );
      pressReturn();
      System.out.print( "        " );
      Traversals.postOrderPrinter( root );
      System.out.println( "\n\n   Now doing an 'in-order' traversal...\n" );
      pressReturn();
      System.out.print( "        " );
      Traversals.inOrderPrinter( root );
      System.out.print( "\n\n\n" );
   }
}

File: Classwork/classwork09/BinaryTree.py
###
# filename: BinaryTree.py
# purpose:  binary tree simulation
# author:   Dr. Johnson
# date:     2023-02-24
###

from BinaryTreeNode import BinaryTreeNode

def main():
   root = BinaryTreeNode( 0 );
   root.add_child( 1, "L" );
   root.add_child( 2, "R" );
   one = root.get_child( "L" );
   two = root.get_child( "R" );
   one.add_child( 3, "L" );
   one.add_child( 4, "R" );
   two.add_child( 5, "L" );
   two.add_child( 6, "R" );
   six = two.get_child( "R" );
   six.add_child( 7, "L" );
   six.add_child( 8, "R" );
   print( "one left: ", one.get_child("L").get_data() )

   print( "\n\n  ==== the tree looks like this [sorta]: \n\n" );
   print( "                  (", root.get_data(), ")" );
   print( "                   / \\" );
   print( "                  /   \\" );
   print( "                 /     \\" );
   print( "                /       \\" );
   print( "               /         \\" );
   print( "              /           \\" );
   print( "             /             \\" );
   print( "          (", one.get_data(), ")           (", two.get_data(), ")" );
   print( "           / \\             / \\" );
   print( "          /   \\           /   \\" );
   print( "         /     \\         /     \\" );
   print( "        /       \\       /       \\" );
   print( "     (", one.get_child("L").get_data(), ")", end="" )
   print( "     (", one.get_child("R").get_data(), ")", end="" )
   print( " (", two.get_child("L").get_data(), ")", end="" )
   print( "     (", two.get_child("R").get_data(), ")" );
   print( "                                / \\" );
   print( "                               /   \\" );
   print( "                            (", six.get_child("L").get_data(), ")", end="" )
   print( " (", six.get_child("R").get_data(), ")" );

main()
